pkg/client/src/app/queries/volumes.ts

```diff
@@ -59,3 +40,54 @@ export const useFetchVolumes = (): IVolumeFetchState => {
     refetch,
   };
 };
+
+export const useCleanRepositoryMutation = ({ onSuccess, onError }) => {
+  const [processId, setProcessId] = useState(null);
+  const [stop, setStop] = useState(false);
+
+  // Mutation to kick off the clean task
+  const { mutate } = useMutation(cleanRepository, {
+    onMutate: () => {
+      setStop(false);
+    },
+    onError: (error) => {
+      console.error(error);
+      setStop(true);
+      onError();
+    },
+    onSuccess: (res) => {
+      setProcessId(res.data.id);
+    },
+  });
+
+  //Fetch until clean task is complete
+  const { isLoading, data } = useQuery(
+    [CleanProgressQueryKey, processId],
+    getTaskById,
+    {
+      onSuccess: (res) => {
+        if (res?.data.state === "Succeeded") {
+          setStop(true);
+          setProcessId(null);
+          onSuccess();
+        } else if (res?.data.state === "Failed") {
+          setStop(true);
+          setProcessId(null);
+          onError();
+        }
+      },
+      onError: (error) => {
+        console.error(error);
+        setStop(true);
+        setProcessId(null);
+        onError();
+      },
+      enabled: processId != null,
+      refetchInterval: stop ? false : 5000,
+      refetchIntervalInBackground: true,
+      refetchOnWindowFocus: false,
+    }
+  );
+
+  return { mutate, data, isLoading };
```

Comment:
Thinking through the state and return values here, I wonder if there's a cleaner way. Right now:
* useCleanRepositoryMutation keeps track of the process ID and whether it should be polling, and returns only the loading state of the polling request (which will be true momentarily every 5 seconds and false otherwise)
* The component above has state for `isCleanDisabled` that should be true while the clean is "running" (it hasn't seen the task succeed/fail yet) and is controlled via the `onSuccess` and `onError` callbacks in here.

Looking at how `useMutation` itself handles status, I wonder if instead you can have this hook own the state of whether the clean process is done (so you can use that as the source of truth for disabling the button instead of having more state up above). I think it could all be driven just from the `processId` state.

Consider this:
* We should only be polling the task when `processId` is not null, so that works for both `enabled` and `refetchInterval` in the useQuery and we don't need the `stop` state.
* We should only disable the "clear repository" button when we're kicking off or polling the task, so we could grab the `isLoading` from the `useMutation` call and call it `isProcessIdLoading`, and this whole hook could return `isCleaning: isProcessIdLoading || !!processId` from here and use that to disable the button instead of having the `isCleanDisabled` state in the component above.

You could still use your `onSuccess` and `onError` to cause the `refetch();` to happen when it's done, but I think `processId` is the only state you need for this whole deal.

Reply:
Actually, holy crap, you might not need any `useState` AT ALL here: the process id is already in the `data` returned by `useMutation`, and you can reset the mutation by calling the `reset` function it returns. So instead of `processId` being a `useState`, you could have `const processId = data?.id || null;`, which will be null until the mutation resolves and go back to being null when you `reset` the mutation. All the state you need is already part of `useMutation`!!